### DISCUSSION TITLE
What are folks doing for OMOP CDM 5.4 observation_period?

### DISCUSSION THREAD

**Discussion by heschmidt04** [chorus-ai/StandardsModule#97] [Structured EHR and ETL Questions] [answered]
Divya had asked me what MGB was doing for code for OMOP observation_period and I realized that the [documentation](https://ohdsi.github.io/CommonDataModel/cdm54.html#observation_period) had a long description with no pointers as to where that information should occur in the ETL process. 
And since MGB is not focusing on payer, cost or plan period - it was just the tables we plan to provide data for. 

Under the documentation in the paragraph ETL conventions it mentions these high level domains and points out _"Condition, Drug, Procedure, Device, Measurement, Visit"_

`It is the discretion of the ETL developer to define these assumptions. In many ETL solutions the start date of the first occurrence or the first high quality occurrence of a Clinical Event (Condition, Drug, Procedure, Device, Measurement, Visit) is defined as the start of the OBSERVATION_PERIOD record, and the end date of the last occurrence of last high quality occurrence of a Clinical Event, or the end of the database period becomes the end of the OBSERVATOIN_PERIOD for each Person.`

So from what I can tell the condition_occurrence, drug_exposure, procedure_occurrence, observation, measurement, device_exposure, visit_occurrence, visit_detail, specimen, and death tables all need to have data and then this table is populated with the person_id and the min start date and max end dates for each person id from those tables. 

It would be interesting to hear what other folks are doing for populating the table with data. 

Best regards,
Heidi

**Comment by jshoughtaling** [comment, marked as answer]
@heschmidt04 - great question and thanks for starting this thread.

The information you quoted is exactly what I have experienced: for the vast majority of ETLs, the Observation Period can simply be defined as the first recorded clinical event for a patient (visit, condition, device, drug, procedure, measurement, observation, specimen, death, etc.) until the last recorded clinical event for that patient. 

> [!WARNING] 
> For those patients who expire during a visit, it's important to use their recorded DEATH datetime as the `observation_period_end_date`, even if other events are recorded after that timepoint. It may also be beneficial to adjust the timestamp of any events recorded post-mortem to the exact DEATH datetime, depending on use case.

Please see this [SQL script](https://github.com/chorus-ai/chorus-container-apps/blob/main/etl/src/etl/merge-etl/observation_period.sql) if you'd like an example of how to implement such logic.
